doc/dev/msgr2.rst

```diff
@@ -235,44 +246,60 @@ Message flow handshake
 In this phase the peers identify each other and (if desired) reconnect to
 an established session.
 
-* TAG_IDENT: identify ourselves::
+* TAG_CLIENT_IDENT (client->server): identify ourselves::
 
-    entity_addrvec_t addr(s)
-    __u8   my type (CEPH_ENTITY_TYPE_*)
+    __le32 num_addrs
+    entity_addrvec_t*num_addrs entity addrs
     __le64 gid (numeric part of osd.0, client.123456, ...)
+    __le64 global_seq
     __le64 features supported (CEPH_FEATURE_* bitmask)
     __le64 features required (CEPH_FEATURE_* bitmask)
     __le64 flags (CEPH_MSG_CONNECT_* bitmask)
-    __le64 cookie (a client identifier, assigned by the sender. unique on the sender.)
```

Comment:
Dropping the cookie here seems like a key change but I don’t think I get it, can you explain?

Reply:
the client doesn't send a cookie; the server does.  (if the client is reconnecting and using the cookie, it uses TAG_RECONNECT.)

Reply:
(i.e., the cookie is one of the main (only?) fields that differ between CLIENT_IDENT and SERVER_IDENT

Reply:
Ah that makes sense; I remembered it being normally unused on the client side but wasn't sure if we always passed this tag over on reconnect.